### PR TITLE
Support navigation group URL

### PR DESCRIPTION
**Pull request by saade**
- [x] Changes have been thoroughly tested to not break existing functionality.
- [x] New functionality has been documented or existing documentation has been updated to reflect changes.
- [x] Visual changes are explained in the PR description using a screenshot/recording of before and after.

This PR adds support for using `->url()` in navigation groups. Using settings as example, using `->url()` in navigation group can redirect me to a page showing all settings available, whereas each navigation item can redirect me to a specific setting.

- If no `->url()` is used, the current collapsible behavior is preserved.
- If `->url()` is used, only the chevron can toggle the group collapsed state.

https://github.com/filamentphp/filament/assets/14329460/3339d08b-3f80-4a6f-b672-be130daadbb1

### CODE REVIEW

**Comment by zepfietje** [comment]
Could you retarget 3.x, @saade?

**Comment by saade** [comment]
> Could you retarget 3.x, @saade?

@danharrin  asked for me to retarget 3.1

https://github.com/filamentphp/filament/pull/8434#issuecomment-1787599347

**Comment by saade** [comment]
@danharrin or @zepfietje let me know if i should retarget 3.x again, otherwise, its ready for review =)

**Comment by danharrin** [comment]
3.1 is correct, thanks

**Comment by zepfietje** [comment]
Could you first merge 3.x into 3.1 then, @danharrin?
I'm missing a couple changes that I made to sidebar groups earlier this week.

**Comment by zepfietje** [comment]
Please request a review from me when done.

**Review by zepfietje — CHANGES_REQUESTED** [review]
Couple of required changes:

- Sidebar group label should only be an `a` tag when it has a URL.
- Sidebar group icon should be inside the link.

Could you work on this and request a new review from me when done, @saade?

**Comment by saade** [comment]
done

File: packages/panels/resources/views/components/sidebar/group.blade.php [hunk 1]
```diff
@@ -22,22 +45,38 @@ class="fi-sidebar-group flex flex-col gap-y-1"
...
 
-            <span
-                class="fi-sidebar-group-label flex-1 text-sm font-semibold leading-6 text-gray-700 dark:text-gray-200"
-            >
-                {{ $label }}
-            </span>
+                    {{ $label }}
+                </a>
+            @else
+                <span class="{{ $labelClasses }}">
```
Comment (zepfietje) [inline packages/panels/resources/views/components/sidebar/group.blade.php]: I'd rather have a Blade conditional in the tag `<{{ $hasUrl ? 'a' : 'span' }}>` than duplicating its contents.

Comment (saade) [reply packages/panels/resources/views/components/sidebar/group.blade.php]: Then for closing the tag too?

```html
<{{ $hasUrl ? 'a' : 'span' }}>
<icon/>
Label
</{{ $hasUrl ? 'a' : 'span' }}>
```

or perhaps

```html
@php($tag = $hasUrl ? 'a' : 'span')
<{{ $tag }}>
<icon/>
Label
</{{ $tag }}>
```

Comment (zepfietje) [reply packages/panels/resources/views/components/sidebar/group.blade.php]: Yep.

Comment (saade) [reply packages/panels/resources/views/components/sidebar/group.blade.php]: okay

Comment (saade) [reply packages/panels/resources/views/components/sidebar/group.blade.php]: @zepfietje done

**Comment by zepfietje** [comment]
I'd like to hold off on this PR until https://github.com/filamentphp/filament/pull/8102 has been implemented. The reason for this being that first-level groups shouldn't be linkable. Have a look at [my comment](https://github.com/filamentphp/filament/pull/8102#issuecomment-1801403794) for context.

**Comment by danharrin** [comment]
Replaced by #9737

**Comment by danharrin** [comment]
Whoops, closed the wrong PR

File: packages/panels/src/Navigation/NavigationGroup.php [hunk 2]
```diff
@@ -75,6 +97,22 @@ public function label(string | Closure | null $label): static
...
+        $this->shouldOpenUrlInNewTab = $condition;
+
+        return $this;
+    }
+
+    public function url(string | Closure | null $url, bool | Closure $shouldOpenInNewTab = false): static
+    {
+        $this->openUrlInNewTab($shouldOpenInNewTab);
+        $this->url = $url;
+        $this->isActiveWhen(fn () => request()->fullUrlIs($url) || request()->path() === trim($url, '/'));
```
Comment (danharrin) [inline packages/panels/src/Navigation/NavigationGroup.php]:
```suggestion
        $this->isActiveWhen(fn (): bool => request()->fullUrlIs($url) || (request()->path() === trim($url, '/')));
```

**Review by danharrin — APPROVED** [review]

**Comment by zepfietje** [comment]
Do we actually need this feature after https://github.com/filamentphp/filament/pull/9737?

**Comment by zepfietje** [comment]
> Do we actually need this feature after #9737?

@saade

**Comment by saade** [comment]
> > Do we actually need this feature after #9737?
> 
> @saade

Hmm, i don't think so, gonna close this and reopen if needed.

**Comment by zepfietje** [comment]
Sure, sounds good.

**Pull request closed without merging**